### ISSUE TITLE
allow .tac files to configure logging behavior

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#3534]
|[<img alt="radix's avatar" src="https://avatars.githubusercontent.com/u/227068?s=50" width="50" height="50">](https://github.com/radix)| @radix reported|
|-|-|
|Trac ID|trac#3534|
|Type|enhancement|
|Created|2008-11-12 23:57:11Z|
|Branch|https://github.com/twisted/twisted/tree/configure-log-observer-in-tacs-3534|

It would be useful to allow .tac files to configure logging behavior by specifying in the .tac file the observer to be used.

Of course, twistd should also have some sort of user interface for customizing logging behavior, which is being tracked in [#11359](https://github.com/twisted/twisted/issues/11359).

<details><summary>Searchable metadata</summary>

```
trac-id__3534 3534
type__enhancement enhancement
reporter__radix radix
priority__normal normal
milestone__ 
branch__branches_configure_log_observer_in_tacs_3534 branches/configure-log-observer-in-tacs-3534
branch_author__radix radix
status__closed closed
resolution__fixed fixed
component__core core
keywords__ 
time__1226534231000000 1226534231000000
changetime__1226856386000000 1226856386000000
version__None None
owner__ 

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="radix's avatar" src="https://avatars.githubusercontent.com/u/227068?s=50" width="50" height="50">](https://github.com/radix)<a name="note_1"></a>|@radix commented|
|-|-|

(In [25356]) Branching to 'configure-log-observer-in-tacs-3534'

**Comment by twisted-trac** [comment]
|[<img alt="radix's avatar" src="https://avatars.githubusercontent.com/u/227068?s=50" width="50" height="50">](https://github.com/radix)<a name="note_2"></a>|@radix removed owner|
|-|-|

Ok, here's basically the same branch as in [#11359](https://github.com/twisted/twisted/issues/11359) but without the --logger stuff.

Please review.

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_3"></a>|@therve set owner to @radix|
|-|-|

1.Some flakes:
```
doc/core/examples/twistd-logging.tac:10: 'sys' imported but unused
twisted/application/app.py:9: 'namedAny' imported but unused
```
 1. In application.xhtml, the example you give is correct, but doesn't bring much against the -l option of twistd. Maybe you could use a `DailyLogFile` to show an advantage?
 1. The twistd-logging.tac example is nice, but it shows bad usage of logging: it doesn't call flush on the log file, and it doesn't use untilConcludes.
 1. The example tac file needs to be referenced in examples/index.xhtml

That's it!

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_4"></a>|@therve commented|
|-|-|

Build results here: http://buildbot.twistedmatrix.com/boxes-supported?branch=branches/configure-log-observer-in-tacs-3534&num_builds=1

**Comment by twisted-trac** [comment]
|[<img alt="radix's avatar" src="https://avatars.githubusercontent.com/u/227068?s=50" width="50" height="50">](https://github.com/radix)<a name="note_5"></a>|@radix removed owner|
|-|-|

Thanks therve. 

 1. fixed
 1. ok, I changed it to use DailyLogFile.
 1. Ok, I made it call flush and use untilConcludes on both the write and flush call, and even explained why.
 1. fixed, and organized a new Logging section in the example index.

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_6"></a>|@therve set owner to @radix|
|-|-|

Awesome, please merge.

**Comment by twisted-trac** [comment]
|[<img alt="radix's avatar" src="https://avatars.githubusercontent.com/u/227068?s=50" width="50" height="50">](https://github.com/radix)<a name="note_7"></a>|@radix set status to `closed`|
|-|-|

(In [25379]) Merge configure-log-observer-in-tacs-3534

Author: radix
Reviewer: therve
Fixes: [#3534](https://github.com/twisted/twisted/issues/3534)

Now it's possible to specify the log observer used by twistd in a .tac
file by setting the ILogObserver component on the application.

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
